File: app/src/main/java/com/chebuso/chargetimer/helpers/PermissionHelper.java
package com.chebuso.chargetimer.helpers;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.support.v4.app.ActivityCompat;

public class PermissionHelper
{
    public static boolean isPermissionsGranted(int[] grantResults) {
        // At least one result must be checked.
        if(grantResults.length < 1){
            return false;
        }

        // Verify that each required permission has been granted, otherwise return false.
        for (int result : grantResults) {
            if (result != PackageManager.PERMISSION_GRANTED) {
                return false;
            }
        }
        return true;
    }

    public static boolean isFullCalendarPermissionsGranted(Context context) {
        return ActivityCompat.checkSelfPermission(context, Manifest.permission.READ_CALENDAR) == PackageManager.PERMISSION_GRANTED
                && ActivityCompat.checkSelfPermission(context, Manifest.permission.WRITE_CALENDAR) == PackageManager.PERMISSION_GRANTED ;
    }
}

File: EvChargeTimer/app/src/main/java/com/chebuso/chargetimer/helpers/PermissionHelper.kt
package com.chebuso.chargetimer.helpers

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import androidx.core.app.ActivityCompat


fun Context.isPermissionGranted(permission: String) =
    ActivityCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED

object PermissionHelper {
    fun isFullCalendarPermissionsGranted(context: Context): Boolean {
        return context.isPermissionGranted(Manifest.permission.READ_CALENDAR)
                && context.isPermissionGranted(Manifest.permission.WRITE_CALENDAR)
    }
}

